src/DataProvider.js

```diff
@@ -1,18 +1,20 @@
 import lo from 'lodash'
 import {cfg} from './config'
+import {Promise} from 'bluebird'
 
-export const RETRY = Symbol('RETRY')
+const RETRY = Symbol('RETRY')
```

Comment:
Since this is used only internally by `getDataWithRetry` it'd be nice to define it right before the function. You can merge the PR then. Also, good job, I love how you've rewritten `getDataWithRetry`.